### PR TITLE
Self-Explaining Neural Networks

### PR DESCRIPTION
**Pull request by pierreelliott**

### CODE REVIEW

File: collections/_posts/2023-07-10-self-explaining-neural-networks.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    title:   "Towards Robust Interpretability with Self-Explaining Neural Networks"
+    venue:   "NeurIPS 2018"
+pdf: https://arxiv.org/pdf/1806.07538.pdf
+---
+
+Github available [here](https://github.com/dmelis/SENN) (PyTorch)
+
+# Highlights
+
+- Post-hoc interpretability methods are not always faithful in regard to model's predictions and their explanations aren't robust to small perturbations of the input[^1]
```
Comment (nathanpainchaud) [inline collections/_posts/2023-07-10-self-explaining-neural-networks.md:17]: Plural:
```suggestion
- Post-hoc interpretability methods are not always faithful in regard to models' predictions and their explanations aren't robust to small perturbations of the input[^1]
```

File: collections/_posts/2023-07-10-self-explaining-neural-networks.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+- The authors propose a class of intrinsically interpretable neural networks models which are as interpretable as linear regression
+
+# Introduction
+
+- Most interpretability methods are post-hoc, explaining the model after its training
+- But their explanations lack stability, small changes to the input can lead to different (or even contradicting) explanations for similar predictions
```
Comment (nathanpainchaud) [inline collections/_posts/2023-07-10-self-explaining-neural-networks.md:23]: _Contradicting_ and _contradictory_ are often mixed up, but they're not interchangeable. In your case you mean to use _contradictory_. See [this question](https://english.stackexchange.com/questions/409244/difference-between-contradictory-and-contradicting) for the differences between the two.
```suggestion
- But their explanations lack stability, small changes to the input can lead to different (or even contradictory) explanations for similar predictions
```

File: collections/_posts/2023-07-10-self-explaining-neural-networks.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+**Locally linear**  
+The function $$\theta(\cdot)$$ can be anything from a simple model to deep neural networks. But to maintain a decent level of interpretability, $$\theta(x)$$ and $$\theta(x')$$ should not differ significantly for close inputs $$x$$ and $$x'$$, $$\theta(\cdot)$$ needs to be stable. More precisely:
+
+$$\nabla_xf(x) \approx \theta(x_0) \tag{3}$$
+
+So the model acts locally, around each $$x_0$$, as a linear model with a vector of stable coefficients $$\theta(x_0)$$
+
+**Concepts**  
+Then, explanations can also be considered in terms of higher order feature, or concepts, derived from the input, like a function $$h(x) : \mathcal{X} \rightarrow \mathcal{Z}$$ where $$\mathcal{Z}$$ is some space of interpretable concepts. The model thus becomes:
```
Comment (nathanpainchaud) [inline collections/_posts/2023-07-10-self-explaining-neural-networks.md:55]: Plural:
```suggestion
Then, explanations can also be considered in terms of higher order features, or concepts, derived from the input, like a function $$h(x) : \mathcal{X} \rightarrow \mathcal{Z}$$ where $$\mathcal{Z}$$ is some space of interpretable concepts. The model thus becomes:
```

File: collections/_posts/2023-07-10-self-explaining-neural-networks.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+# Limitations
+
+- No details on model performances, even in supplementary materials
+- Definition of interpretable concepts is a problem in itself
+- The final prediction is more explainable, but the intermediate steps are not: the concept encoder and relevance parametrizer are still complete black-boxes
+
+# References
+
+[^1]: [Alvarez-Melis D. and Jaakkola T.S., On the Robustness of Interpretability Methods. ]()
```
Comment (nathanpainchaud) [inline collections/_posts/2023-07-10-self-explaining-neural-networks.md:127]: The reference here seems incomplete and the link itself is missing.

**Review by nathanpainchaud — CHANGES_REQUESTED** [review]
Great review, and excellent level of English :slightly_smiling_face: There are just a few phrasing/typos to correct!

File: collections/_posts/2023-07-10-self-explaining-neural-networks.md (hunk 4 above)
Comment (pierreelliott) [reply collections/_posts/2023-07-10-self-explaining-neural-networks.md:127]: Woups, I've done it too quickly at the last minute. I'll correct it!

**Pull request merged**